artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/DockerContainer_ContainerManagementInterface-w1/source/src/main/java/org/opentosca/artifacttemplates/dockercontainer/DockerContainer.java

```diff
@@ -0,0 +1,178 @@
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.ExecCreateCmdResponse;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientBuilder;
+import com.github.dockerjava.core.command.ExecStartResultCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DockerContainer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerContainer.class);
+
+    private final DockerClient client;
+    private final String containerId;
+
+    public DockerContainer(String dockerEngineURL, String dockerEngineCertificate, String containerId) {
+        this.client = getClient(dockerEngineURL, dockerEngineCertificate);
+        this.containerId = containerId;
+    }
+
+    public void awaitAvailability() throws InterruptedException {
+        long startTime = System.currentTimeMillis();
+        long endTime = startTime + 25000;
+        InterruptedException error = null;
+        while (System.currentTimeMillis() < endTime) {
+            try {
+                // if we can execute pwd without issues ssh is up!
+                execCommand("pwd");
+                LOG.info("Container is available");
+                return;
+            } catch (InterruptedException e) {
+                LOG.error("Could not await availability", e);
+                error = e;
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e1) {
+                    LOG.error("Could not await availability", e1);
+                }
+            }
+        }
+        if (error != null) {
+            throw error;
+        }
+    }
+
+    public String execCommand(String command) throws InterruptedException {
+        LOG.info("Executing shell command: {}", command);
+
+        try (OutputStream stream = getStringOutputStream()) {
+            ExecCreateCmdResponse execCreateCmdResponse = client.execCreateCmd(containerId)
+                    .withTty(true)
+                    .withAttachStdout(true)
+                    .withCmd("/bin/bash", "-c", command)
+                    .exec();
+
+            client.execStartCmd(execCreateCmdResponse.getId())
+                    .withTty(true)
+                    .withDetach(false)
+                    .exec(new ExecStartResultCallback(stream, System.err))
+                    .awaitCompletion();
+
+            String result = stream.toString();
+            LOG.info("Execution log:\n {}", result);
+            return result;
+        } catch (IOException e) {
+            LOG.error("Could not execute command", e);
+        }
+
+        return "";
+    }
+
+    // TODO: replace this with other new function
+    public String replaceHome(String commandString, boolean all) {
```

Comment:
I dont understand the logic of replaceHome.
It produces e.g.  `//` at the beginning ...

```
opentosca-engine-ia-jdk17-1  | Replaced ~ with user home ('/'): 'sleep 1 && mkdir -p ~/RealWorld-Application_Angular-Spring-w1.csar/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/NGINX_latest-Install-w1/files'
opentosca-engine-ia-jdk17-1  | 2022-03-15 07:37:08.250  INFO 14 --- [nio-8080-exec-2] o.o.a.dockercontainer.DockerContainer    : Executing shell command: sleep 1 && mkdir -p //RealWorld-Application_Angular-Spring-w1.csar/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/NGINX_latest-Install-w1/files
```